### PR TITLE
feat (worker): get git info, create var if necessary

### PR DESCRIPTION
**Pull request by yesnault**
Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

**Comment by yesnault** [comment]
Tested with git >= version 1.7.10.4 

git clone log:

capture d ecran 2017-12-15 a 23 38 58


And created vars on a child pipeline: 

capture d ecran 2017-12-15 a 23 35 32

File: engine/worker/builtin_gitclone.go [hunk 1]
```diff
@@ -237,3 +246,100 @@ func runGitClone(w *currentWorker) BuiltInAction {
...
+
+			if _, err := w.addVariableInPipelineBuild(gitMessage, params); err != nil {
+				return fmt.Errorf("Error on addVariableInPipelineBuild (message): %s", err)
+			}
+			sendLog(fmt.Sprintf("git.message: %s", info.Message))
+		} else {
+			sendLog(fmt.Sprintf("git.message: %s", message))
+		}
+
+		if message == "" {
```
Comment (fsamin) [inline engine/worker/builtin_gitclone.go]: author

**Comment by fsamin** [comment]
are the variables available in the current pipeline, in the next steps or stage for instance ?

**Comment by yesnault** [comment]
Next step in same job -> yes, it's available
Next job in next stages -> ok too

File: engine/worker/builtin_gitclone.go [hunk 2]
```diff
@@ -237,3 +246,100 @@ func runGitClone(w *currentWorker) BuiltInAction {
...
+
+			if _, err := w.addVariableInPipelineBuild(gitAuthor, params); err != nil {
+				return fmt.Errorf("Error on addVariableInPipelineBuild (author): %s", err)
+			}
+			sendLog(fmt.Sprintf("git.author: %s", info.Author))
+		} else {
+			sendLog(fmt.Sprintf("git.author: %s", author))
+		}
+
+		gitAuthorEmail := sdk.Variable{
```
Comment (sguiheux) [inline engine/worker/builtin_gitclone.go]: IMprove repository webhook to add it

**Pull request merged**